### ISSUE TITLE
Zmiana parametrów w ofercie

### ISSUE THREAD

**Issue by Lebioda** [allegro/allegro-api#4800]
Skróciliście parametr "Kod producenta" z 50 na 30 znaków co zablokowało jakąkolwiek edycję ofert.

Jak ten parametr zmienić przez API?

Próbuje przez 

`PATCH /sale/product-offers/9872723570`

wysłać:

`{
product: {
parameters: [
{
id: "219809",
values: [
"Prowadnica do dłut i strugów"
]
}
]
}
}`

ale nic się nie zmienia...

**Comment by MaciejFrackowiak** [comment]
W przypadku tablic np. parametry, zdjęcia, opis oferty musisz przekazać wszystkie dane, które w danej tablicy się znadują - czyli w przypadku parametrów musisz przekazać wszystkie dotychczasowe i edytowane łącznie.

**Comment by sikorskim** [comment]
Jest jakiś sposób na masową zmianę pola "Kod producenta"? W moim przypadku te pole będzie takie same jak SKU, ale w przypadku każdego produktu id tego parametru jest inne.

**Comment by MaciejFrackowiak** [comment]
@sikorskim Nie, nie ma takiej możliwości. Każda oferta musi zostać wyedytowana pojedynczo.

**Comment by Slpw20** [comment]
Zauważyłem, że wspomniany kod producenta jest różny dla stabilnej wersji (limit 40 znaków), a dla wersji z sandboxa (limit 50 znaków) - przykładowo dla kategorii **_Elektronika >> Komputery >> Urządzenia sieciowe >> Akcesoria >> Kable >> Patchcord_**. Proszę o informację, czemu występuje dana rozbieżność?

Dodatkowo proszę o informację, czy wspomniana zmiana limitu znaków dla danego pola była wspominana w jednym z artykułów / wiadomości? Jeżeli tak to proszę o link, ponieważ nie mogę go zlokalizować.

**Comment by AureliuszBrussy** [comment]
Listę kategorii i parametrów na Sandboxie aktualizujemy raz na kwartał, dlatego mogą pojawić się różnice pomiędzy produkcją, a środowiskiem testowym. 

Nie informujemy w newsach o zmianach w kategoriach i parametrach (czy to o np. ich usunięciu, czy o właśnie zmianie limitu znaków).

**Comment by Lebioda** [comment]
> W przypadku tablic np. parametry, zdjęcia, opis oferty musisz przekazać wszystkie dane, które w danej tablicy się znadują - czyli w przypadku parametrów musisz przekazać wszystkie dotychczasowe i edytowane łącznie.

Mogę prosić o jakiś przykład od A do Z, czyli odczytanie parametrów z oferty, zmiana jednego (np. kod produktu) i wrzucenie z powrotem na Allegro?

Ciągle coś nie pasuje. Nie można nawet odesłać parametrów bez zmian, bo i tak rzucacie errorami, np. _Given parameter id `237190:Brand` should no have been passed as `offer` parameter._

Prosta z nazwy funkcjonalność a skomplikowaliście ją na potęgę...

**Comment by AureliuszBrussy** [comment]
Nie mam takiego kompletnego przykładu, natomiast błąd o którym piszesz wskazuje, że próbujesz edytować parametr produktowy,  przesyłając go w części ofertowej żądania (poza polem **product.parameters**, czyli po prostu w **parameters**). W [naszym poradniku](https://developer.allegro.pl/product_offers/#jak-ustawi%c4%87-parametry-ofertowe) znajdziesz informację, jak rozpoznać rodzaj parametru. 

Dodatkowo zwróć uwagę na to, czy produkt, z którym powiązana jest oferta ma już przypisaną wartość do danego parametru. Jeżeli tak - edytowanie takiego parametru również zakończy się niepowodzeniem ze względu na to, że nie będzie zgodny z wartością w produkcie. Swobodnie możesz natomiast edytować np. parametry nieobowiązkowe, niewypełnione w danym produkcie.
